### PR TITLE
unbias MMD grad

### PR DESCRIPTION
**Pull request by josephbowles**
The previous qubit implementation of the MMD loss functions used lax.stop_gradient in order to avoid a division by zero when the std of the estimator was zero. This makes the MMD gradient biased, since gradients should flow through this line. This can all be avoided by returning the variance in the qubit expval function rather than the standard deviation, and puts it in line with the qudit version, which returns the complex covariance matrix. 

The U-statistic to unbias the qudit MMD loss was also overly complicated. The code was using an optional return variable `mean_y_squared`, but the same can be achieved using the covariance matrix, so the code was simplified to do that. 

**Description of the Change:**
- standard deviation changed to variance in qubit expval function
- lax.stop_gradient removed from MMD loss functions
- mean_y_squared optional return removed from qudit expval function 

**Benefits:**
simpler code. MMD gradient now unbiased. 

**Possible Drawbacks:**
none as far as I can see

### CODE REVIEW

File: pennylane/labs/tests/tcdq/test_qudit_mmd_loss.py [hunk 1]
```diff
@@ -272,14 +272,15 @@ def test_matching_moments_near_zero(self):
 
         inner = l_obs.astype(jnp.float64) @ data.astype(jnp.float64).T
         data_moments = jnp.mean(jnp.exp(2j * jnp.pi * inner / d), axis=1)
+        variances = (1 - jnp.abs(data_moments) ** 2) / (n_samples - 1)
```
Comment (willjmax) [inline pennylane/labs/tests/tcdq/test_qudit_mmd_loss.py:275]: Should this be `(1 - jnp.abs(data_moments))**2`? We're assuming the mean is 1?

Comment (josephbowles) [reply pennylane/labs/tests/tcdq/test_qudit_mmd_loss.py:275]: ok I checked this out and it's legit. It follows from an algebraic manipulation of the sample variance for unit norm complex random variables.

**Review by austingmhuang — APPROVED** [review]
Thanks @josephbowles

**Review by willjmax — APPROVED** [review]
